coderd/database/dbauthz/dbauthz.go

```diff
@@ -2307,6 +2344,13 @@ func (q *querier) GetTemplateParameterInsights(ctx context.Context, arg database
 	return q.db.GetTemplateParameterInsights(ctx, arg)
 }
 
+func (q *querier) GetTemplatePresetsWithPrebuilds(ctx context.Context, templateID uuid.NullUUID) ([]database.GetTemplatePresetsWithPrebuildsRow, error) {
+	if err := q.authorizeContext(ctx, policy.ActionRead, rbac.ResourceTemplate); err != nil {
```

Comment:
self-review: "prebuilds" is important to note here. this should belong to the prebuilds user

Reply:
An unfortunate consequence for how we have things, you should just do a `q.GetTemplateByID(ctx, tv.TemplateID.UUID)`. That does the `read` check on the template. 

It adds another DB query to this, and is basically unavoidable.

Reply:
@Emyrk Now it checks:
```go
if err := q.authorizeContext(ctx, policy.ActionRead, rbac.ResourceSystem); err != nil {
	return nil, err
}
```